File: Java_dsa/Misc/ReverseAndCheckForPalindrome.java
public class ReverseAndCheckForPalindrome {

    public static int reverseNum(int num){
        int reverseNum = 0;
        while (num != 0) {
            reverseNum = reverseNum * 10 + (num % 10);
            num /= 10;
        }
        return reverseNum;
    }

    public static boolean checkNumberIspalindrome(int num){
        int revNum = ReverseAndCheckForPalindrome.reverseNum(num);
        if(revNum == num) {
            System.out.println(num + " is palindrome");
            return true;
        }else{
            System.out.println(num + " is not a palindrome");
            return false;
        }
    }
    public static void main(String[] args) {
        int num = 45654;
        ReverseAndCheckForPalindrome.checkNumberIspalindrome(num);
    }
}

File: Javascript_dsa/Misc/ReverseAndCheckForPalindrome.js
const reverseNumber = (num) => {
    let reversedNum = 0;
    while (num != 0) {
        reversedNum = reversedNum * 10 + (num % 10)
        num  = Math.floor(num/10);
    }
    return reversedNum;
}

const checkNumIsPalindrome = (num)=>{
    let originalNum = num;
    let revNum = reverseNumber(num)
    if(originalNum === revNum) return `${num} is palindrome`
    if(originalNum !== revNum) return `${num} is not palindrome`
}

console.log(checkNumIsPalindrome(43210234))